lockfile/src/parsers/pypi.rs

```diff
@@ -151,16 +177,56 @@ fn identifier_list(input: &str) -> Result<&str, &str> {
 }
 
 fn line_done(input: &str) -> Result<&str, &str> {
-    let (input, _) = space0(input)?;
+    // Allow for spaces and arguments not impacting resolution.
+    let (input, _) = recognize(many0(alt((nl_space1, package_hash))))(input)?;
+
     eof(input)
 }
 
+/// Parse package hashes.
+///
+/// Example:
+///   --hash=sha256:
+/// 8c2f9abd47a9e8df7f0c3f091ce9497d011dc3b31effcf4c85a6e2b50f4114ef
+fn package_hash(input: &str) -> Result<&str, &str> {
+    // Argument name.
+    let (input, _) = tag("--hash=")(input)?;
+
+    // Hash variant.
+    let (input, _) = alphanumeric1(input)?;
+
+    // Separator.
+    let (input, _) = tag(":")(input)?;
+
+    // Package hash.
+    alphanumeric1(input)
+}
+
 /// A combinator that takes a parser `inner` and produces a parser that also
 /// consumes both leading and trailing whitespace, returning the output of
 /// `inner`.
 fn ws<'a, F>(inner: F) -> impl FnMut(&'a str) -> Result<&str, &str>
 where
     F: Fn(&'a str) -> Result<&str, &str>,
 {
-    delimited(space0, inner, space0)
+    delimited(nl_space0, inner, nl_space0)
+}
+
+/// Newline-aware space0.
+///
+/// This automatically handles " \\\n" and treats it as normal space.
+fn nl_space0(input: &str) -> Result<&str, &str> {
+    recognize(many0(alt((space1, line_continuation))))(input)
+}
+
+/// Newline-aware space1.
+///
+/// This automatically handles " \\\n" and treats it as normal space.
+fn nl_space1(input: &str) -> Result<&str, &str> {
+    recognize(many1(alt((space1, line_continuation))))(input)
+}
+
+/// Recognize line continuations.
+fn line_continuation(input: &str) -> Result<&str, &str> {
+    recognize(tuple((tag("\\"), opt(tag("\r")), tag("\n"))))(input)
```

Comment:
It doesn't make much difference, but this could be written with `line_ending`

```suggestion
    recognize(tuple((tag("\\"), line_ending)))(input)
```